core/plugins/org.polarsys.capella.core.data.migration/src/org/polarsys/capella/core/data/migration/contribution/ImagePathInRichTextAttributeContribution.java

```diff
@@ -0,0 +1,318 @@
+import java.text.MessageFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.emf.common.util.TreeIterator;
+import org.eclipse.emf.ecore.EAttribute;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EStructuralFeature;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.sirius.business.api.image.Base64ImageHelper;
+import org.eclipse.sirius.business.api.image.ImageManager;
+import org.eclipse.sirius.business.api.image.RichTextAttributeRegistry;
+import org.eclipse.sirius.business.api.query.ResourceQuery;
+import org.eclipse.sirius.ext.emf.edit.EditingDomainServices;
+import org.polarsys.capella.common.ef.ExecutionManager;
+import org.polarsys.capella.core.data.migration.Activator;
+import org.polarsys.capella.core.data.migration.context.MigrationContext;
+import org.polarsys.capella.core.model.handler.command.CapellaResourceHelper;
+
+/**
+ * This contribution applies on the attribute containing rich text(html). It will </br>
+ * - replace the image base64 encoded string by a path to a local file that is also created.</br>
+ * - If the absolute path leads to an image, update the absolute path to a relative path and copy the image and the
+ * "project"/images folder
+ * 
+ * @author lfasani
+ */
+public class ImagePathInRichTextAttributeContribution extends AbstractMigrationContribution {
+  private static final String HTML_IMAGE_ABSOLUTE_PATH_PATTERN = "<img.*?src=\"(file:/.*?)\".*?/>"; //$NON-NLS-1$
+
+  private static final String IMAGE_NAME_FORMAT = "yyyyMMdd_HHmmssSSS"; //$NON-NLS-1$
+
+  @Override
+  public void postMigrationExecute(ExecutionManager executionManager, ResourceSet resourceSet,
+      MigrationContext context) {
+    //@formatter:off
+    List<Resource> resources = resourceSet.getResources().stream()
+        .filter(res -> {
+          return CapellaResourceHelper.isCapellaResource(res) || new ResourceQuery(res).isAirdOrSrmResource(); 
+        }).filter(res -> {
+          //keep only the resource corresponding to the context otherwise the migration may be done many times for some resources.
+          IFile file = ResourcesPlugin.getWorkspace().getRoot().getFile(new org.eclipse.core.runtime.Path(res.getURI().toPlatformString(true)));
+          IFile contextFile = context.getResource();
+          return contextFile!=null && contextFile.equals(file);
+        }).collect(Collectors.toList());
```

Comment:
Not sure about this loop
resources seems always returning one resource if .capella or .aird/srm
`Resource resource = resourceSet.getResource(EcoreUtil2.getURI(context.getResource()), false); ` ?

Reply:
done